platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdmin.java

```diff
@@ -427,18 +433,32 @@ public void updateForLocation(String pid, String location,
             throw new IOException("Argument configurationTable cannot be null");
         }
 
-        // sanity check to make sure no values are
-        // null
-        for (Entry<String, Object> curEntry : configurationTable.entrySet()) {
-            Object value = curEntry.getValue();
-            if (value == null || (value instanceof ArrayList && CollectionUtils.sizeIsEmpty(value))) {
-                curEntry.setValue("");
+        // find the config's metatype so we can check cardinality
+        Configuration config = configurationAdmin.getConfiguration(pid, location);
+        List<Map<String, Object>> services = listServices();
+        List<Map<String, Object>> tempMetatype = null;
+        for (Map<String, Object> service : services) {
+            String id = (String) service.get(ConfigurationAdminExt.MAP_ENTRY_ID);
+            if (id.equals(config.getPid()) || (id.equals(config.getFactoryPid())
+                    && (boolean) service.get(ConfigurationAdminExt.MAP_FACTORY))) {
+                tempMetatype = (List<Map<String, Object>>) service
+                        .get(ConfigurationAdminExt.MAP_ENTRY_METATYPE);
             }
         }
 
-        Dictionary<String, Object> configurationProperties = new Hashtable<String, Object>(
-                configurationTable);
-        Configuration config = configurationAdmin.getConfiguration(pid, location);
+        final List<Map<String, Object>> metatype = tempMetatype;
+        List<Map.Entry<String, Object>> configEntries = new ArrayList<>();
+        configEntries.addAll(configurationTable.entrySet());
+        if (metatype == null) {
+            throw new IllegalArgumentException("Could not find metatype for " + pid);
+        }
+        // now we have to filter each property based on its cardinality
+        CollectionUtils.transform(configEntries, new CardinalityTransformer(metatype, pid));
+
+        Dictionary<String, Object> configurationProperties = new Hashtable<>();
+        for (Map.Entry<String, Object> configEntry : configEntries) {
+            configurationProperties.put(configEntry.getKey(), configEntry.getValue());
```

Comment:
The transformer is supposed to prevent null values, and a null key is both invalid usage and would blow up before that line. Wouldn't checking here be redundant?